mautrix_facebook/web/segment_analytics.py

```diff
@@ -0,0 +1,33 @@
+import logging
+import os
+
+from yarl import URL
+import aiohttp
+
+from ..user import User
+
+log = logging.getLogger("mau.web.public.analytics")
+http: aiohttp.ClientSession
+host: str = "api.segment.io"
+
+try:
+    segment_key = os.environ["SEGMENT_API_KEY"]
+    http = aiohttp.ClientSession()
+    log.debug("Enabled segment analytics tracking")
+except (ImportError, KeyError):
+    segment_key = None
+
+
+def track(user: User, event: str, properties: dict) -> None:
```

Comment:
This needs to be async and all the calls need to be awaited (or `asyncio.create_task()`ed) too

```suggestion
async def track(user: User, event: str, properties: dict) -> None:
```

Although if you go the `asyncio.create_task` route so the user wouldn't have to wait for segment to respond, it might make more sense to change `track` into a wrapper function (then the calls don't need to be awaited):

```python
def track(...):
    if segment_key:
        asyncio.create_task(_track(...))

async def _track(...):
    ...

Reply:
👍 yep that's nicer will change to that.

Reply:
https://github.com/mautrix/facebook/pull/213/commits/2a1e29a7297ddf2ad74dc724d3c2e4090c15d826